File: task3/c++/Subtask-2.cpp
#include <fstream>
#include <iostream>

int main() {
    std::ifstream inputFile("input.txt");
    std::ofstream outputFile("output.txt");

    if (!inputFile.is_open()) {
        std::cerr << "Error opening input file." << std::endl;
        return 1;
    }
    if (!outputFile.is_open()) {
        std::cerr << "Error opening output file." << std::endl;
        return 1;
    }

    std::string line;
    while (std::getline(inputFile, line)) {
        outputFile << line << std::endl;
    }

    if (inputFile.eof()) {
        std::cout << "End of file reached successfully." << std::endl;
    } else if (inputFile.fail()) {
        std::cerr << "Error reading from input file." << std::endl;
    }

    inputFile.close();
    outputFile.close();

    std::cout << "Program completed successfully." << std::endl;
    return 0;
}

File: task3/Rust/Subtask2.rs
use std::fs::File;
use std::io::{self, Read, Write};

fn main() -> io::Result<()> {
    // Open the input file
    let mut input_file = File::open("input.txt")?;

    // Create a buffer to store the file's content
    let mut content = String::new();

    // Read the content of the input file into the buffer
    input_file.read_to_string(&mut content)?;

    // Open (or create) the output file
    let mut output_file = File::create("output.txt")?;

    // Write the content to the output file
    output_file.write_all(content.as_bytes())?;

    println!("Content copied from input.txt to output.txt successfully!");

    Ok(())
}
